sentry-android-core/src/main/java/io/sentry/android/core/SystemEventsBreadcrumbsIntegration.java

```diff
@@ -92,46 +115,168 @@ public void register(final @NotNull IScopes scopes, final @NotNull SentryOptions
             this.options.isEnableSystemEventBreadcrumbs());
 
     if (this.options.isEnableSystemEventBreadcrumbs()) {
+      addLifecycleObserver(this.options);
+      registerReceiver(this.scopes, this.options, /* newIntegration = */ true);
+    }
+  }
 
-      try {
-        options
-            .getExecutorService()
-            .submit(
-                () -> {
-                  try (final @NotNull ISentryLifecycleToken ignored = startLock.acquire()) {
-                    if (!isClosed) {
-                      startSystemEventsReceiver(scopes, (SentryAndroidOptions) options);
+  private void registerReceiver(
+      final @NotNull IScopes scopes,
+      final @NotNull SentryAndroidOptions options,
+      final boolean newIntegration) {
```

Comment:
nit, to make it a bit more clear what this flag is about
```suggestion
      final boolean reportAsNewIntegration) {
```